scanpy/tools/_leiden.py

```diff
@@ -143,13 +147,19 @@ def leiden(
     # as this allows for the accounting of a None resolution
     # (in the case of a partition variant that doesn't take it on input)
     if use_weights:
-        partition_kwargs["weights"] = np.array(g.es["weight"]).astype(np.float64)
-    partition_kwargs["n_iterations"] = n_iterations
-    partition_kwargs["seed"] = random_state
+        clustering_args["weights"] = "weight" if use_igraph else np.array(g.es["weight"]).astype(np.float64)
+    clustering_args["n_iterations"] = n_iterations
+    if not use_igraph:
+        clustering_args["seed"] = random_state
```

Comment:
I think both of these should be added to `uns` as outputs, `weights` in the `params` part of the dict and `use_igraph` at the top levels

```python
adata['leiden']
# { 'params': { ... 'use_weights': True }, 'use_igraph': False }